### PR TITLE
Sorr task615 postprocessing docsend data

### PR DESCRIPTION
**Pull request by shab95**
- [Found Zapier connection from Docsend to google sheets](https://zapier.com/apps/docsend/integrations/google-sheets/175058/create-google-sheet-rows-for-new-visits-in-docsend)
- Processed csv with two possible routes
     - GSheets API
     - Local CSV import
- Found top 10 unique emails sorted by highest duration on the document

### CODE REVIEW

**Comment by shab95** [comment]
Hey @samarth9008 

Not sure why these fast tests are failing, looks like something is wrong with the docker library I am using as the tests are running fine on my side.

I tried updating the docker image, merged in master, and updated docker, but found no luck. Whenever you find some free time, I'd appreciate some help. In the mean time, I'll look into some other solutions.

**Review by samarth9008 — COMMENTED** [review]
You can remove the code and instructions related to csv. Not needed IMO

File: linkedin/notebooks/Postprocess_Docsend_Data.ipynb [hunk 1]
```diff
@@ -0,0 +1,438 @@
+{
```
Comment (gpsaggese) [inline linkedin/notebooks/Postprocess_Docsend_Data.ipynb:1]: @samarth9008 let's reorg a bit the dirs (like under `marketing` one dir per tool).
This should go under `marketing/docsend/notebooks`

**Review by gpsaggese — COMMENTED** [review]
Zapier is interesting but we don't really need in real-time.
Can we trigger a full dump through Zapier?
Otherwise it's ok to log in and dump the data.

File: linkedin/notebooks/Postprocess_Docsend_Data.ipynb (hunk 1 above)
Comment (shab95) [reply linkedin/notebooks/Postprocess_Docsend_Data.ipynb:1]: For this, is it ok if I just move the .ipynb and .py files to this directory?

File: linkedin/notebooks/Postprocess_Docsend_Data.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+## clean data
+df["Duration"] = pd.to_timedelta(df["Duration"])
+df = df.replace(r"^\s*$", np.nan, regex=True)
+df
+
+# %%
+## compute data
+sorted_by_duration_df = df.sort_values(by="Duration", ascending=False)
+sorted_by_duration_df = sorted_by_duration_df.dropna(subset=["Email"])
+top_ten_emails = sorted_by_duration_df["Email"].unique()[:10]
```
[the remark this answers is not in the document]
Comment (shab95) [reply linkedin/notebooks/Postprocess_Docsend_Data.py:74]: are the emails in a list good for this?  This is what it results at the moment:
```
['info@electriccapital.com' 'jschmidt@a16z.com' 'jayhao1@gmail.com'
 'dteten@versatilevc.com' 'james@fika.vc' 'info@rre.com'
 'cf@foxventures.io' 'eesa.ahmad@crypto.com' 'dennis@rre.com'
 'jackdavis@eureliosollutions.com' 'pitch@twosigmaventures.com'
 'info@dncapital.com' 'pr@gsrventures.com'
 'investments@rockawaycapital.com' 'membership@heavybit.com'
 'greg.ballanco@coinbase.com' 'kwqc@hvznypecdvhnwbk.fod' 'info@parafi.com'
 'info@delphidigital.io' 'info@metastablecapital.com'
 'crypto@crypto-kaizen.com' 'info@wellington.com'
 'office@barclayventures.com' 'info@rockawaycapital.com'
 'information@madrona.com' 'michael@quiet.com' 'info@gumi-cryptos.com'
 'info@willcapllc.com' 'tiffany.galvin-cohen@point72.com'
 'info@connectventures.com' 'info@shasta.vc' 'info@tcg.co'
 'hello@icodrops.com' 'info_benelux@okta.com' 'info@unshackledvc.com'
 'jack.statza@av.vc' 'contact.kvg@kakaocorp.com' 'info@thrivecap.com'
 'gary@lauderpartners.com' 'admin@nextplaycapital.com']
 ```

**Comment by shab95** [comment]
I don't think dumping rows is an option. I believe that Zapier doesn't have any triggers and actions for this scenario. I think the only Docsend triggers are: 
- New link created
- New space created
- New space download
- New visit

File: linkedin/notebooks/Postprocess_Docsend_Data.ipynb (hunk 1 above)
Comment (samarth9008) [reply linkedin/notebooks/Postprocess_Docsend_Data.ipynb:1]: Yes move the files to `marketing/docsend/notebooks`

File: linkedin/notebooks/Postprocess_Docsend_Data.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+# # Description
+#
+# This notebook was used for prototyping ways to postprocess docsend data that has been exported to a Google Sheet in GDrive.
+#
+# - getting the docsend data using this zapier connection might be good if used properly in conjunction with the GSheets API: https://zapier.com/apps/docsend/integrations/google-sheets/175058/create-google-sheet-rows-for-new-visits-in-docsend
+
+# %% [markdown]
+# ## Instructions
+#
+# - Run pip install gspread-pandas in terminal
```
[the remark this answers is not in the document]
Comment (samarth9008) [reply linkedin/notebooks/Postprocess_Docsend_Data.py:25]: This is not still resolved.

File: linkedin/notebooks/Postprocess_Docsend_Data.py [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+# - Follow the directions here for *Client Credentials* and follow the *Service Account Route* : https://gspread-pandas.readthedocs.io/en/latest/getting_started.html
+#     - Remember to save your credentials to the gspread_pandas folder
+# - Get the email of your service account
+#     - This can be found under Service Accounts header in the Credentials Tab of your project
+# - Share your Google Sheets file with the service account email(press Share top right)
+# - Rename the gsheets_name variable to the name of the file
```
Comment (samarth9008) [inline linkedin/notebooks/Postprocess_Docsend_Data.py:31]: -> gsheets_name(In code format)

File: linkedin/notebooks/Postprocess_Docsend_Data.py [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+# %%
+# imports
+
+import numpy as np
+import pandas as pd
+from gspread_pandas import Spread
+# !sudo /bin/bash -c "(source /venv/bin/activate; pip install gspread-pandas)"
```
Comment (samarth9008) [inline linkedin/notebooks/Postprocess_Docsend_Data.py:39]: Add this a different cell above imports and run it. Remove the comment from here and instruction from above.

File: linkedin/notebooks/Postprocess_Docsend_Data.ipynb (hunk 1 above)
Comment (shab95) [reply linkedin/notebooks/Postprocess_Docsend_Data.ipynb:1]: There is a folder called `marketing/notebooks`, I wasn't too sure to put it in there or make a new directory with `marketing/docsend/notebooks`, but I ended up just making this new directory and dropping these files in this new directory.

File: marketing/docsend/notebooks/Postprocess_Docsend_Data.py [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+#
+# - Follow the directions here for *Client Credentials* and follow the *Service Account Route* : https://gspread-pandas.readthedocs.io/en/latest/getting_started.html
+#     - Remember to save your credentials to the gspread_pandas folder
+# - Get the email of your service account
+#     - This can be found under Service Accounts header in the Credentials Tab of your project
+# - Share your Google Sheets file with the service account email(press Share top right)
+# - Rename the `gsheets_name` variable to the name of the file
+
+# %%
+# Install gspread-pandas
```
Comment (samarth9008) [inline marketing/docsend/notebooks/Postprocess_Docsend_Data.py:33]: Add dot in the end.

File: marketing/docsend/notebooks/Postprocess_Docsend_Data.py [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+!sudo /bin/bash -c "(source /venv/bin/activate; pip install gspread-pandas)"
+
+# %%
+# Imports
```
Comment (samarth9008) [inline marketing/docsend/notebooks/Postprocess_Docsend_Data.py:37]: You can move this markdown heading

File: marketing/docsend/notebooks/Postprocess_Docsend_Data.py [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+import numpy as np
+import pandas as pd
+from gspread_pandas import Spread
+
+# %%
+## Google Sheet name
```
Comment (samarth9008) [inline marketing/docsend/notebooks/Postprocess_Docsend_Data.py:43]: Single hash and dot in the end. Merge it with `#Load data section`

File: marketing/docsend/notebooks/Postprocess_Docsend_Data.py [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+gsheets_name = "Kaizen - VC presentation - v1.5-export"
+
+# %%
+## Load data
```
Comment (samarth9008) [inline marketing/docsend/notebooks/Postprocess_Docsend_Data.py:47]: Make this a markdown heading

File: marketing/docsend/notebooks/Postprocess_Docsend_Data.py [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+spread = Spread(gsheets_name)
+data = spread.sheets[0].get_values()
+headers = data.pop(0)
+df = pd.DataFrame(data, columns=headers)
+
+# %%
+## Clean data
```
Comment (samarth9008) [inline marketing/docsend/notebooks/Postprocess_Docsend_Data.py:54]: Make this a markdown heading

File: marketing/docsend/notebooks/Postprocess_Docsend_Data.py [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+df["Duration"] = pd.to_timedelta(df["Duration"])
+df = df.replace(r"^\s*$", np.nan, regex=True)
+df
+
+# %%
+## Compute data
```
Comment (samarth9008) [inline marketing/docsend/notebooks/Postprocess_Docsend_Data.py:60]: Make this a markdown heading

File: marketing/docsend/notebooks/Postprocess_Docsend_Data.py [hunk 12]
```diff
@@ -0,0 +1,67 @@
...
+sorted_by_duration_df = df.sort_values(by="Duration", ascending=False)
+sorted_by_duration_df = sorted_by_duration_df.dropna(subset=["Email"])
+sorted_emails = sorted_by_duration_df["Email"].unique()
+
+# %%
+## Show data
+print(sorted_emails)
```
Comment (samarth9008) [inline marketing/docsend/notebooks/Postprocess_Docsend_Data.py:67]: Merge it with compute data

File: marketing/docsend/notebooks/Postprocess_Docsend_Data.py [hunk 13]
```diff
@@ -0,0 +1,71 @@
...
+# %% [markdown]
+# # Clean data
+
+# %%
+df["Duration"] = pd.to_timedelta(df["Duration"])
+df = df.replace(r"^\s*$", np.nan, regex=True)
+df
+
+# %% [markdown]
+# ## Compute data
```
Comment (samarth9008) [inline marketing/docsend/notebooks/Postprocess_Docsend_Data.py:65]: I think it has an extra hash.

File: marketing/docsend/notebooks/Postprocess_Docsend_Data.py [hunk 14]
```diff
@@ -0,0 +1,71 @@
...
+import pandas as pd
+from gspread_pandas import Spread
+
+# %% [markdown]
+# # Load data
+
+# %%
+# Google Sheet name.
+gsheets_name = "Kaizen - VC presentation - v1.5-export"
+
```
Comment (samarth9008) [inline marketing/docsend/notebooks/Postprocess_Docsend_Data.py:50]: No blank line

**Review by samarth9008 — APPROVED** [review]
LGTM

**Pull request merged**